### DISCUSSION TITLE
Set label for ID field

### DISCUSSION THREAD

**Discussion by woylie** [ory/kratos#2368] [Q&A] [answered]
I have an `emails` field configured like this:

```json
        "emails": {
          "type": "array",
          "title": "E-Mail",
          "items": {
            "type": "string",
            "title": "E-Mail",
            "format": "email",
            "ory.sh/kratos": {
              "credentials": {
                "password": {
                  "identifier": true
                },
                "totp": {
                  "account_name": true
                }
              },
              "verification": {
                "via": "email"
              }
            }
          }
        },
```

When I initialize a browser login flow (`GET /self-service/login/browser`), the UI node for the email field has the label `ID`. Is there a way to set the label to `E-mail` in the schema? There are no other identifiers in the schema. I'd prefer not to manually override the label in my UI builder.

**Comment by aeneasr** [comment, marked as answer]
We don’t know which identity will be used beforehand. As you can have many identities with different schemas it is not possible to determine what the label of the ID node will be before we know the schema of the identity that is trying to sign in. So using the label ID is a compromise here!

**Reply by woylie** [reply]
That makes sense. I wonder if it would make sense to add something like an `id_label` option that can be set for cases where all identity schemas use the same identifier type?

```yaml
selfservice:
  flows:
    login:
      id_label: "E-Mail"
```

**Reply by aeneasr** [reply]
I think that could be an acceptable solution. One issue here would be that people probably want more and more configuration flexibility (what type of input it is, translation, ...) and it will become just too much configuration. We have a follow-up issue tracked here: https://github.com/ory/kratos/issues/2255
